crates/shared/src/persistence/sqlite.rs

```diff
@@ -0,0 +1,211 @@
+
+        // Convert SystemTime to a format SQLite understands (RFC 3339)
+        let target_timestamp = DateTime::<Utc>::from(target_time)
+            .naive_utc()
+            .format("%Y-%m-%d %H:%M:%S")
+            .to_string();
+
+        let rows = sqlx::query(
+            r#"
+            SELECT id, tx_data, created_at FROM transactions
+            WHERE created_at <= ? ;
+            "#,
+        )
+        .bind(target_timestamp)
+        .fetch_all(&self.pool)
+        .await?;
+
+        let tx_data_list = rows
+            .into_iter()
+            .map(|row| row.get::<Vec<u8>, _>("tx_data"))
+            .collect();
+        Ok(tx_data_list)
+    }
+
+    async fn remove(&self, tx_data: Vec<u8>) -> Result<(), sqlx::Error> {
+        let result = sqlx::query(
+            r#"
+            DELETE FROM transactions WHERE tx_data = ?;
+            "#,
+        )
+        .bind(tx_data)
+        .execute(&self.pool)
+        .await?;
+
+        if result.rows_affected() > 0 {
+            Ok(())
+        } else {
+            Err(sqlx::Error::RowNotFound)
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::get_sqlite_test_db_path;
+    use super::BuilderPersistence;
+    use super::SqliteTxnDb;
+    use std::time::Instant;
+
+    /// This test checks we can set up sqlite properly
+    /// and can do basic append() and load()
+    #[tokio::test]
+    async fn test_persistence_append_and_load_txn() {
+        // Initialize the database
+        tracing::debug!(
+            "get_sqlite_test_db_path() = {:?}",
+            get_sqlite_test_db_path()
+        );
+        let db = SqliteTxnDb::new(get_sqlite_test_db_path()).await.expect(
+            "In test_persistence_append_and_load_txn, it should be able to initiate a sqlite db.",
+        );
+
+        db.clear()
+            .await
+            .expect("In test_persistence_remove_txn, it should be able to clear all transactions.");
+
+        // Append a few transactions
+        let test_tx_data_list = vec![vec![1, 2, 3], vec![4, 5, 6]];
+        for tx in test_tx_data_list.clone() {
+            db.append(tx).await.expect("In test_persistence_append_and_load_txn, there shouldn't be any error when doing append");
+        }
+
+        // Set timeout_after to the current time
+        let timeout_after = Instant::now();
+
+        // Simulate some delay
+        tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+
+        // Append one more transaction
+        db.append(vec![7, 8, 9]).await.expect("In test_persistence_append_and_load_txn, there shouldn't be any error when doing append");
+
+        // Load transactions before timeout_after
+        let tx_data_list = db.load(timeout_after).await.expect(
+            "In test_persistence_append_and_load_txn, it should be able to load some transactions.",
+        );
+        tracing::debug!("Transaction data before timeout: {:?}", tx_data_list);
+
+        assert_eq!(tx_data_list, test_tx_data_list);
```

Comment:
`load(self, _instant_)` would give us all the transactions created at some point before the given _instant_. Sorry about the confusion from naming. The test is just to check it could behave correctly, i.e. if we have some transactions submitted after the given _instant_, they shouldn't be returned, and only those submitted before should be returned.

`load(self, _instant_)` is designed this way because we have a function `collect_txns(timeout_after)` to collect all the submitted txn till an instant. Now I think for database it might be the same to return all the transactions, as the only `timeout_after` we'll set is near future, and this is our only use case. I'd vote to remove it if there's more overhead due to this argument or else I'd keep it. How do you think?